### ISSUE TITLE
TCP Connection reset by peer post Cilium installation

### ISSUE THREAD

**Issue by krishadvani** [cilium/cilium#21119]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

After deploying Cilium(1.12.1) with AWS CNI chaining, we find ourselves in a situation where our requests are being forwarded to pods which are currently not in the ready state. 

Our scenario is that we have 3 Istio Gateway pods running under the istio-system namespace, which, if scaled up, causes connection reset errors as indicated below:

`read tcp 10.132.28.36:53860->172.20.137.154:80: read: connection reset by peer`

Here 10.132.28.36:53860 is the client and 172.20.137.154:80 is the IP of the SVC of the istio-gateway, which is a ClusterIP.

We ran a tcpdump on the machine on which this pod is running, and we indeed get a RST flag from the istio-system pod, which is currently on a 0/1 status.

The Cilium Endpoint shows up as ready instantly, as soon as the pod goes into Init or container creation stage. 

We tried running Hubble cli to observe for any dropped traffic, but we didn't receive any "dropped" requests

What we have already tried:

1. Ensure that AWS CNI is up to date and above the recommended version.
2. Updated Kube-proxy and core DNS to the latest version.
3. Restarted all unmanaged pods.
4. Verified no invalid entries are part of the table
5. Ran cilium tests to ensure the cluster is setup correctly

Removing Cilium from the cluster resolves the problem.

Is there anything we are doing wrong? 

### Cilium Version

v1.11.5

### Kernel Version

Linux ip-10-129-29-21.ap-south-1.compute.internal 5.4.181-99.354.amzn2.x86_64 #1 SMP Wed Mar 2 18:50:46 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Kubernetes Version

1.21

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by krishadvani** [comment]
Providing an update on the above:

We installed Cilium without AWS chaining, it didn't solve the problem, however, deleting Kube-proxy and going with a strict replacement policy for Kube proxy fixed this issue. We are able to replicate this issue in brand new clusters as well, hence we are pretty sure that there is something wrong either with the default configuration of cilium as per the documentation, or this is a bug that hasn't been previously reported.

**Comment by pchaigno** [comment]
Do you know why the Istio pod says it's not ready? Did you check why it returns a RST (not listening on port yet maybe)?

Could you please share a sysdump of the cluster?

**Comment by krishadvani** [comment]
The issue persisted with non istio pods as well. We found that pods that were not ready yet, i.e before the readiness probe has successfully completed would register a an endpoint in cilium and end up forwarding a request to a newly launched pod.

Unfortunately, we have moved from AWS chaining mode with Cilium which makes this issue hard to reproduce for us at the moment, we shall re-open the ticket with more information when we test cilium with AWS chaining.

We sincerely appreciate your time.
